compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/JvmOptimizationLowering.kt

```diff
@@ -148,121 +146,59 @@ class JvmOptimizationLowering(val context: JvmBackendContext) : FileLoweringPass
                     if (left.isNullConst() && right is IrConst<*> || right.isNullConst() && left is IrConst<*>)
                         return IrConstImpl.constFalse(expression.startOffset, expression.endOffset, context.irBuiltIns.booleanType)
 
-                    val safeCallLeft = parseSafeCall(left)
-                    if (safeCallLeft != null && right.type.isJvmPrimitive()) {
-                        return rewriteSafeCallEqeqPrimitive(safeCallLeft, right, expression)
-                    }
-
-                    val safeCallRight = parseSafeCall(right)
-                    if (safeCallRight != null && left.type.isJvmPrimitive()) {
-                        return rewritePrimitiveEqeqSafeCall(left, safeCallRight, expression)
+                    if (expression.symbol == context.irBuiltIns.eqeqSymbol) {
+                        if (right.type.isJvmPrimitive()) {
```

Comment:
Please, add a test with `@EnhancedNullability Int` (`@NotNull Integer getInteger();` in Java).